# AttachmentsColored local overrides vs RT 5.0.10 defaults
# Generated (UTC): 2026-10-12T09:00:29Z
# Baseline: docca/RT5_10/rt-5.0.10/share/html
# Local: docca/Plugins/RT-Extension-AttachmentsColored-rt5/html
# Direction: minus = share default; plus = local override
# Exact relative-path matches only; html/Callbacks/ excluded.
# CRLF normalized to LF; all other whitespace preserved.
# Snapshot comparison only: not a history of upstream changes.
# Matched: 4; changed: 4; identical: 0; no default: 115
#
# NO DEFAULT: html/Admin/NextCloudFiles/index.html
# NO DEFAULT: html/Ajax/AddNewTicket
# NO DEFAULT: html/Ajax/AddToTicket
# NO DEFAULT: html/Ajax/AjaxDeleteCategory
# NO DEFAULT: html/Ajax/AjaxModifyCategory
# NO DEFAULT: html/Ajax/AjaxModifyComment
# NO DEFAULT: html/Ajax/AjaxModifyFolder
# NO DEFAULT: html/Ajax/AjaxSendMQOpen
# NO DEFAULT: html/Ajax/AjaxUploadAttachment
# NO DEFAULT: html/Ajax/AjaxUploadGDrive
# NO DEFAULT: html/Ajax/AjaxUploadNextCloud
# NO DEFAULT: html/Ajax/AjaxUploadOneDrive
# NO DEFAULT: html/Ajax/LoadAttachments.json
# NO DEFAULT: html/Ajax/LoadNodes.json
# CHANGED: html/Elements/SelectTimezone
# NO DEFAULT: html/Helpers/Toggle/AttachmentsColored_off
# NO DEFAULT: html/NoAuth/css/cloud.css
# NO DEFAULT: html/NoAuth/css/dataTables.bootstrap4.min.css
# NO DEFAULT: html/NoAuth/css/dataTables.checkboxes.css
# NO DEFAULT: html/NoAuth/css/jquery.dataTables.css
# NO DEFAULT: html/NoAuth/js/English.json
# NO DEFAULT: html/NoAuth/js/FileSaver.min.js
# NO DEFAULT: html/NoAuth/js/Hungarian.json
# NO DEFAULT: html/NoAuth/js/Russian.json
# NO DEFAULT: html/NoAuth/js/dataTables.bootstrap4.min.js
# NO DEFAULT: html/NoAuth/js/dataTables.buttons.min.js
# NO DEFAULT: html/NoAuth/js/dataTables.checkboxes.min.js
# NO DEFAULT: html/NoAuth/js/dataTables.responsive.min.js
# NO DEFAULT: html/NoAuth/js/dataTables.select.min.js
# NO DEFAULT: html/NoAuth/js/datetime-moment.js
# NO DEFAULT: html/NoAuth/js/file-size.js
# NO DEFAULT: html/NoAuth/js/i18n/af.js
# NO DEFAULT: html/NoAuth/js/i18n/ar.js
# NO DEFAULT: html/NoAuth/js/i18n/az.js
# NO DEFAULT: html/NoAuth/js/i18n/bg.js
# NO DEFAULT: html/NoAuth/js/i18n/bn.js
# NO DEFAULT: html/NoAuth/js/i18n/bs.js
# NO DEFAULT: html/NoAuth/js/i18n/ca.js
# NO DEFAULT: html/NoAuth/js/i18n/cs.js
# NO DEFAULT: html/NoAuth/js/i18n/da.js
# NO DEFAULT: html/NoAuth/js/i18n/de.js
# NO DEFAULT: html/NoAuth/js/i18n/dsb.js
# NO DEFAULT: html/NoAuth/js/i18n/el.js
# NO DEFAULT: html/NoAuth/js/i18n/en.js
# NO DEFAULT: html/NoAuth/js/i18n/es.js
# NO DEFAULT: html/NoAuth/js/i18n/et.js
# NO DEFAULT: html/NoAuth/js/i18n/eu.js
# NO DEFAULT: html/NoAuth/js/i18n/fa.js
# NO DEFAULT: html/NoAuth/js/i18n/fi.js
# NO DEFAULT: html/NoAuth/js/i18n/fr.js
# NO DEFAULT: html/NoAuth/js/i18n/gl.js
# NO DEFAULT: html/NoAuth/js/i18n/he.js
# NO DEFAULT: html/NoAuth/js/i18n/hi.js
# NO DEFAULT: html/NoAuth/js/i18n/hr.js
# NO DEFAULT: html/NoAuth/js/i18n/hsb.js
# NO DEFAULT: html/NoAuth/js/i18n/hu.js
# NO DEFAULT: html/NoAuth/js/i18n/hy.js
# NO DEFAULT: html/NoAuth/js/i18n/id.js
# NO DEFAULT: html/NoAuth/js/i18n/is.js
# NO DEFAULT: html/NoAuth/js/i18n/it.js
# NO DEFAULT: html/NoAuth/js/i18n/ja.js
# NO DEFAULT: html/NoAuth/js/i18n/ka.js
# NO DEFAULT: html/NoAuth/js/i18n/km.js
# NO DEFAULT: html/NoAuth/js/i18n/ko.js
# NO DEFAULT: html/NoAuth/js/i18n/lt.js
# NO DEFAULT: html/NoAuth/js/i18n/lv.js
# NO DEFAULT: html/NoAuth/js/i18n/mk.js
# NO DEFAULT: html/NoAuth/js/i18n/ms.js
# NO DEFAULT: html/NoAuth/js/i18n/nb.js
# NO DEFAULT: html/NoAuth/js/i18n/ne.js
# NO DEFAULT: html/NoAuth/js/i18n/nl.js
# NO DEFAULT: html/NoAuth/js/i18n/pl.js
# NO DEFAULT: html/NoAuth/js/i18n/ps.js
# NO DEFAULT: html/NoAuth/js/i18n/pt-BR.js
# NO DEFAULT: html/NoAuth/js/i18n/pt.js
# NO DEFAULT: html/NoAuth/js/i18n/ro.js
# NO DEFAULT: html/NoAuth/js/i18n/ru.js
# NO DEFAULT: html/NoAuth/js/i18n/sk.js
# NO DEFAULT: html/NoAuth/js/i18n/sl.js
# NO DEFAULT: html/NoAuth/js/i18n/sq.js
# NO DEFAULT: html/NoAuth/js/i18n/sr-Cyrl.js
# NO DEFAULT: html/NoAuth/js/i18n/sr.js
# NO DEFAULT: html/NoAuth/js/i18n/sv.js
# NO DEFAULT: html/NoAuth/js/i18n/th.js
# NO DEFAULT: html/NoAuth/js/i18n/tk.js
# NO DEFAULT: html/NoAuth/js/i18n/tr.js
# NO DEFAULT: html/NoAuth/js/i18n/uk.js
# NO DEFAULT: html/NoAuth/js/i18n/vi.js
# NO DEFAULT: html/NoAuth/js/i18n/zh-CN.js
# NO DEFAULT: html/NoAuth/js/i18n/zh-TW.js
# NO DEFAULT: html/NoAuth/js/jquery.dataTables.min.js
# NO DEFAULT: html/NoAuth/js/jszip-utils.min.js
# NO DEFAULT: html/NoAuth/js/jszip.min.js
# NO DEFAULT: html/NoAuth/js/select2.full.js
# NO DEFAULT: html/NoAuth/js/select2.full.min.js
# NO DEFAULT: html/NoAuth/js/select2.js
# NO DEFAULT: html/NoAuth/js/select2.min.js
# NO DEFAULT: html/NoAuth/js/util_AttachmentsColored.js
# CHANGED: html/Ticket/Elements/AddAttachments
# NO DEFAULT: html/Ticket/Elements/RT-Extension-EmailSablon-rt5/README.md
# NO DEFAULT: html/Ticket/Elements/RT-Extension-EmailSablon-rt5/etc/EmailSablon_Config.pm
# NO DEFAULT: html/Ticket/Elements/RT-Extension-EmailSablon-rt5/html/Ajax/ContentUpdate
# NO DEFAULT: html/Ticket/Elements/RT-Extension-EmailSablon-rt5/html/Ajax/SablonAttachments
# NO DEFAULT: html/Ticket/Elements/RT-Extension-EmailSablon-rt5/html/Callbacks/EmailSablon/Elements/Header/Head
# NO DEFAULT: html/Ticket/Elements/RT-Extension-EmailSablon-rt5/html/Callbacks/EmailSablon/Elements/MessageBox/Default
# NO DEFAULT: html/Ticket/Elements/RT-Extension-EmailSablon-rt5/html/Callbacks/EmailSablon/Ticket/Create.html/BeforeMessageBox
# NO DEFAULT: html/Ticket/Elements/RT-Extension-EmailSablon-rt5/html/Callbacks/EmailSablon/Ticket/Update.html/BeforeMessageBox
# NO DEFAULT: html/Ticket/Elements/RT-Extension-EmailSablon-rt5/html/NoAuth/css/emailsablon.css
# NO DEFAULT: html/Ticket/Elements/RT-Extension-EmailSablon-rt5/html/NoAuth/js/util_EmailSablon.js
# NO DEFAULT: html/Ticket/Elements/RT-Extension-EmailSablon-rt5/initialdata
# NO DEFAULT: html/Ticket/Elements/RT-Extension-EmailSablon-rt5/lib/RT/Extension/EmailSablon.pm
# NO DEFAULT: html/Ticket/Elements/RT-Extension-EmailSablon-rt5/po/hu.po
# CHANGED: html/Ticket/Elements/ShowAttachments
# NO DEFAULT: html/Ticket/Elements/ShowAttachmentsOriginal
# CHANGED: html/Ticket/Elements/ShowSummary
# NO DEFAULT: html/Ticket/Elements/adatlap_sablon_u6_1.xml
# NO DEFAULT: html/Ticket/Elements/adatlap_sablon_u6_2.xml
# NO DEFAULT: html/Update/ShowCommentUpdate
# NO DEFAULT: html/Update/ShowFolderUpdate

--- share/html/Elements/SelectTimezone
+++ local/html/Elements/SelectTimezone
@@ -1,75 +1,12 @@
-%# BEGIN BPS TAGGED BLOCK {{{
-%#
-%# COPYRIGHT:
-%#
-%# This software is Copyright (c) 1996-2026 Best Practical Solutions, LLC
-%#                                          <sales@bestpractical.com>
-%#
-%# (Except where explicitly superseded by other copyright notices)
-%#
-%#
-%# LICENSE:
-%#
-%# This work is made available to you under the terms of Version 2 of
-%# the GNU General Public License. A copy of that license should have
-%# been provided with this software, but in any event can be snarfed
-%# from www.gnu.org.
-%#
-%# This work is distributed in the hope that it will be useful, but
-%# WITHOUT ANY WARRANTY; without even the implied warranty of
-%# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
-%# General Public License for more details.
-%#
-%# You should have received a copy of the GNU General Public License
-%# along with this program; if not, write to the Free Software
-%# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
-%# 02110-1301 or visit their web page on the internet at
-%# http://www.gnu.org/licenses/old-licenses/gpl-2.0.html.
-%#
-%#
-%# CONTRIBUTION SUBMISSION POLICY:
-%#
-%# (The following paragraph is not intended to limit the rights granted
-%# to you to modify and distribute this software under the terms of
-%# the GNU General Public License and is only of importance to you if
-%# you choose to contribute your changes and enhancements to the
-%# community by submitting them to Best Practical Solutions, LLC.)
-%#
-%# By intentionally submitting any modifications, corrections or
-%# derivatives to this work, or any other work intended for use with
-%# Request Tracker, to Best Practical Solutions, LLC, you confirm that
-%# you are the copyright holder for those contributions and you grant
-%# Best Practical Solutions,  LLC a nonexclusive, worldwide, irrevocable,
-%# royalty-free, perpetual, license to use, copy, create derivative
-%# works based on those contributions, and sublicense and distribute
-%# those contributions and any derivatives thereof.
-%#
-%# END BPS TAGGED BLOCK }}}
-<%ONCE>
-use DateTime;
-
-my @names = DateTime::TimeZone->all_names;
-
-my %label;
-my $dt = DateTime->now;
-for ( @names ) {
-    $dt->set_time_zone( $_ );
-    $label{$_} = $_ . ' ' . $dt->strftime('%z');
-}
-
-</%ONCE>
-<select class="form-control selectpicker" name="<% $Name %>">
-% if ( $ShowNullOption ) {
-<option value=""><&|/l&>System Default</&> (<% RT->Config->Get('Timezone') %>)</option>
-% }
-% foreach my $tz (@names) {
-<option value="<% $tz %>" <% ($Default||'') eq $tz? 'selected="selected"' :''
- |n %> ><% $label{$tz} || $tz %></option>
+<select name="<% $Name %>" class="form-control selectpicker">
+% for my $tz (@names) {
+  <option value="<% $tz %>" <% $tz eq $Default ? 'selected="selected"' : '' |n %>><% $tz %></option>
 % }
 </select>
 <%ARGS>
-$ShowNullOption => 1
-$Name => undef
-$Default => 0
+$Name => 'Timezone'
+$Default => undef
 </%ARGS>
-
+<%INIT>
+my @names = ('Europe/Budapest');
+</%INIT>
--- share/html/Ticket/Elements/AddAttachments
+++ local/html/Ticket/Elements/AddAttachments
@@ -2,7 +2,7 @@
 %#
 %# COPYRIGHT:
 %#
-%# This software is Copyright (c) 1996-2026 Best Practical Solutions, LLC
+%# This software is Copyright (c) 1996-2021 Best Practical Solutions, LLC
 %#                                          <sales@bestpractical.com>
 %#
 %# (Except where explicitly superseded by other copyright notices)
@@ -61,13 +61,13 @@
 </div>
 % } # end of if
 
+<div class="form-group">
 % if ($FullWidth) {
-<div class="form-row">
-  <div class="col-12">
+  <div class="form-row">
+    <div class="col-12">
 % }
         <div class="dropzone" id="attach-dropzone">
         </div>
-
 <script type="text/javascript">
 var MaxAttachmentSizeMiB = RT.Config.MaxAttachmentSize/(1024*1024);
 
@@ -168,9 +168,13 @@
 });
 </script>
         <div class="hidden old-attach">
-          <div class="custom-file">
-            <input type="file" id="Attach" name="Attach" class="custom-file-input" />
-            <label class="custom-file-label" for="Attach"><&|/l&>Choose file</&></label>
+          <div class="form-row">
+            <div class="col-12">
+              <div class="custom-file">
+                <input type="file" id="Attach" name="Attach" class="custom-file-input" />
+                <label class="custom-file-label" for="Attach"><&|/l&>Choose file</&></label>
+              </div>
+            </div>
           </div>
           <div class="form-row">
             <div class="col-12">
@@ -178,17 +182,18 @@
             </div>
           </div>
         </div>
-
 % if ($FullWidth) {
+    </div>
   </div>
-</div>
 % }
-
+</div>
 % if ($HasExisting) {
 <div class="form-row">
-  <div class="label col-3"><&|/l&>Include attachments</&>:</div>
-  <div class="value col-9" id="reuse-attachments">
-    <& /Ticket/Elements/ShowAttachments,
+  <div class="col-12"><&|/l&>Include attachments</&>:</div>
+</div>
+<div class="form-row">
+  <div class="value col-12" id="reuse-attachments">
+    <& /Ticket/Elements/ShowAttachmentsOriginal,
       Ticket       => $TicketObj,
       Selectable   => 1,
       HideTitleBox => 1,
--- share/html/Ticket/Elements/ShowAttachments
+++ local/html/Ticket/Elements/ShowAttachments
@@ -2,7 +2,7 @@
 %#
 %# COPYRIGHT:
 %#
-%# This software is Copyright (c) 1996-2026 Best Practical Solutions, LLC
+%# This software is Copyright (c) 1996-2018 Best Practical Solutions, LLC
 %#                                          <sales@bestpractical.com>
 %#
 %# (Except where explicitly superseded by other copyright notices)
@@ -45,84 +45,924 @@
 %# those contributions and any derivatives thereof.
 %#
 %# END BPS TAGGED BLOCK }}}
-<&| /Widgets/TitleBox, title => loc('Attachments'), 
-        title_class=> 'inverse',  
-        class => 'ticket-info-attachments',
-        color => "#336699",
-        hide_chrome => $HideTitleBox &>
 
-% $m->callback( %ARGS, CallbackName => 'BeforeList', TicketObj => $Ticket, Attachments => $Attachments, Documents => \%documents, IsChecked => \%is_checked, ShowMore => \$show_more );
+<!-- NEW TICKET -->
+<div class="modal fade" id="myModal" tabindex="-1" role="dialog">
+  <div class="modal-dialog" role="document">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h5 class="modal-title" id="exampleModalLabel"><&|/l&>New ticket</&></h5>
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+      </div>
+      <div class="modal-body">
+        <form>
+          <div class="form-group">
+            <label class="col-form-label"><%loc('Subject')%>:</label>
+            <input type="text" name="subj" id="subj" class="form-control" />
+          </div>
+          <div class="form-group">
+            <label class="col-form-label"><%loc('Queue')%>:</label>
+% my $queue_selector = $m->scomp('/Elements/SelectNewTicketQueue', Name => 'Queue1', AutoSubmit => 0, Placeholder => loc('Queue') );
+<% $queue_selector |n %>
+          </div>
+          <div class="form-group">
+            <label class="col-form-label"><%loc('User')%>:</label>
+% my $user_selector = $m->scomp('/Elements/SelectOwner', Name => 'Owner1', DefaultValue => 0, Default => $session{'CurrentUser'}->id );
+<% $user_selector |n %>
+          </div>
+          <div class="form-group">
+            <label for="message-text" class="col-form-label"><&|/l&>Message</&>:</label>
+            <textarea class="form-control" id="message-text1"></textarea>
+%#            <span><&|/l&>Attachments</&>:</span>
+            <span id="att_list"></span>
+          </div>
+        </form>
+      </div>
+      <div class="modal-footer">
+        <button type="button" class="btn btn-secondary" data-dismiss="modal"><&|/l&>Close</&></button>
+        <button type="button" class="btn btn-primary" id="send1"><&|/l&>Create ticket</&></button>
+      </div>
+    </div>
+  </div>
+</div>
+<!-- /NEW TICKET -->
+
+<!-- COMMENT TO TICKET -->
+<div class="modal fade" id="myModal2" tabindex="-1" role="dialog">
+  <div class="modal-dialog" role="document">
+    <div class="modal-content">
+      <div class="modal-header">
+        <h5 class="modal-title" id="exampleModalLabel"><&|/l&>Comment</&></h5>
+        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+          <span aria-hidden="true">&times;</span>
+        </button>
+      </div>
+      <div class="modal-body">
+        <form>
+          <div class="form-group">
+            <label class="col-form-label"><%loc('Ticket id')%>:</label>
+            <input type="text" class="form-control" id="ticketid">
+          </div>
+          <div class="form-group">
+            <label for="message-text" class="col-form-label"><&|/l&>Message</&>:</label>
+            <textarea class="form-control" id="message-text2"></textarea>
+%#            <span><&|/l&>Attachments</&>:</span>
+            <span id="att_list2"></span>
+          </div>
+        </form>
+      </div>
+      <div class="modal-footer">
+        <button type="button" class="btn btn-secondary" data-dismiss="modal"><&|/l&>Close</&></button>
+        <button type="button" class="btn btn-primary" id="send2"><&|/l&>Comment</&></button>
+      </div>
+    </div>
+  </div>
+</div>
+<!-- /COMMENT TO TICKET -->
 
-<div class="attachment-list">
+<&| /Widgets/TitleBox, title => loc('Attachments'), 
+		title_class=> 'inverse',  
+		class => 'ticket-info-attachments',
+		color => "#336699",
+		hide_chrome => $HideTitleBox &>
 
-% foreach my $key (sort { lc($a) cmp lc($b) } keys %documents) {
+% $m->callback( %ARGS, CallbackName => 'BeforeList', TicketObj => $Ticket, Attachments => $Attachments, Documents => \%documents, IsChecked => \%is_checked, ShowMore => \$show_more );
 
-<span class="filename"><%$key%></span>
-<ul class="list-group list-group-compact" <% $Selectable ? 'class="selectable"' : '' |n %> >
-% foreach my $rev (@{$documents{$key}}) {
-% if ($rev->ContentLength) {
-<li class="list-group-item">
+<!-- <table id="attachments" class="display nowrap" width="100%"> -->
+<div class="attachments-table-wrapper">
+<table id="attachments" class="table table-striped table-bordered table-hover" width="100%">
+	<thead>
+		<tr>
+%#			<th></th>
+			<th style="width:30px;"><input type='checkbox' id='checkAll' onclick='checkAllClick(this);' /></th>
+			<th style="min-width:80px;max-width:120px;"><&|/l&>Highlight</&></th>
+			<th><&|/l&>Filename</&></th>
+			<th style="min-width:70px;max-width:90px;"><&|/l&>Date</&></th>
+			<th style="min-width:70px;max-width:100px;"><&|/l&>User</&></th>
+			<th style="min-width:50px;max-width:70px;"><&|/l&>Size</&></th>
+			<th style="min-width:100px;max-width:180px;"><&|/l&>Tagging</&></th>
+			<th style="min-width:80px;max-width:150px;"><&|/l&>comment</&></th>
+		</tr>
+	</thead>
+<!--
+	<tfoot>
+		<tr>
+			<th><&|/l&>Highlight</&></th>
+			<th><&|/l&>Filename</&></th>
+			<th><&|/l&>Date</&></th>
+			<th><&|/l&>User</&></th>
+			<th><&|/l&>Size</&></th>
+			<th><&|/l&>Tagging</&></th>
+			<th><&|/l&>comment</&></th>
+		</tr>
+	</tfoot>
+//-->
 
-  <div class="custom-control custom-checkbox">
+</table>
+</div>
 % if ($Selectable) {
-      <input type="checkbox" class="custom-control-input" id="AttachExisting-<% $rev->Id %>" name="AttachExisting" value="<% $rev->Id %>" \
-        <% $is_checked{$rev->Id} ? 'checked' : '' %>
-      />
+<div id="attachment-existing-selected" hidden></div>
 % }
 
-<label <% $Selectable ? 'class="custom-control-label"' : '' |n%> for="AttachExisting-<% $rev->Id %>">
-<span class="revision">
-% if (my $url = RT->System->ExternalStorageURLFor($rev)) {
-<a href="<%$url%>">
-% } else {
-<a href="<%RT->Config->Get('WebPath')%>/<% $DisplayPath %>/Attachment/<%$rev->TransactionId%>/<%$rev->Id%>/<%$rev->Filename | un %>">
-% }
-% my $desc = loc("[_1] ([_2]) by [_3]", $rev->CreatedAsString, $rev->FriendlyContentLength, $m->scomp('/Elements/ShowUser', User => $rev->CreatorObj));
-<% $desc |n%>
-</a>
-</span>
-    </label>
-  </div>
-</li>
-% }
-% }
-</ul>
+<script src="<%$RT::WebPath%>/NoAuth/js/jquery.dataTables.min.js" type="text/javascript"></script>
+<script src="<%$RT::WebPath%>/NoAuth/js/dataTables.bootstrap4.min.js" type="text/javascript"></script>
+<script src="<%$RT::WebPath%>/NoAuth/js/file-size.js" type="text/javascript"></script>
+<script src="<%$RT::WebPath%>/NoAuth/js/dataTables.select.min.js" type="text/javascript"></script>
+<!-- <script src="<%$RT::WebPath%>/NoAuth/js/datetime-moment.js" type="text/javascript"></script> -->
+<!-- <script src="<%RT->Config->Get('WebPath')%>/NoAuth/js/dataTables.responsive.min.js" type="text/javascript"></script> -->
+<script src="<%RT->Config->Get('WebPath')%>/NoAuth/js/dataTables.buttons.min.js" type="text/javascript"></script>
+
+<!-- download zip -->
+<script src="<%RT->Config->Get('WebPath')%>/NoAuth/js/jszip.min.js" type="text/javascript"></script>
+<script src="<%RT->Config->Get('WebPath')%>/NoAuth/js/jszip-utils.min.js" type="text/javascript"></script>
+<script src="<%RT->Config->Get('WebPath')%>/NoAuth/js/FileSaver.min.js" type="text/javascript"></script>
 
+<script type="text/javascript">
+
+	var colors = <%to_json(\%colormap, {canonical=>1})|n%>;
+	var dt, result;
+	var bShowHide = false;
+	var selectedRows = {}; // Track selected rows across all pages
+	var selectableAttachments = <% $Selectable ? 'true' : 'false' %>;
+	var checkedAttachmentIds = <% to_json([keys %is_checked]) |n %>;
+	var initialAttachmentIds = <% to_json([map { $_->Id } map { @$_ } values %documents]) |n %>;
+	var limitAttachments = <% defined($Count) ? 'true' : 'false' %>;
+
+	jQuery(function ($) {
+		$(document).ready( function () {
+
+			function formatResult(node) {
+				var $result = $('<span>').attr('title', node.lib || '').css('padding-left', (20 * (node.level - 1)) + 'px');
+				$result.append($('<img>').attr('src', <% RT->Config->Get('WebPath') . '/static/images/icons8-folder-16.png' |n,j %>));
+				$result.append(document.createTextNode(' ' + (node.text || '')));
+				return $result;
+			};
+			result = $.ajax({
+				type: "POST",
+				url: "<%RT->Config->Get('WebPath')%>/Ajax/LoadAttachments.json",
+				data: { 'ticketId': '<%$Ticket->Id%>' },
+				success: function(data) {
+					var allAttachments = data.attachments;
+					var checkedIds = checkedAttachmentIds.map(String);
+					var initialIds = initialAttachmentIds.map(String);
+					allAttachments.forEach(function(attachment) {
+						if (checkedIds.indexOf(String(attachment.attid)) !== -1) {
+							selectedRows[attachment.attid] = { attid: attachment.attid, link: attachment.checkbox };
+						}
+					});
+					var displayedAttachments = limitAttachments ? allAttachments.filter(function(attachment) {
+						return initialIds.indexOf(String(attachment.attid)) !== -1;
+					}) : allAttachments;
+					dt = $('#attachments').DataTable({
+						language: {
+							url: '<%$RT::WebPath%>/NoAuth/js/Hungarian.json'
+						},
+						pageLength: 50,
+						lengthMenu: [ [10, 25, 50, 100, -1], [10, 25, 50, 100, "<&|/l&>All</&>"] ],
+						dom: 'Bl<"__toolbar">frtip',
+						initComplete: function() {
+							$("div.__toolbar").html('&nbsp;&nbsp;&nbsp;&nbsp;<label><input id="showhide" name="showhide" type="checkbox" value="1" onclick="hiddenShowHide( this.checked ); console.log( \'check box \' + this.checked); return true;">&nbsp;<&|/l&>Show hidden files</&></label>');
+%#							$('.selectnode').each(function() {
+%#								for (var i = 0; i < data.attachments.length; i++){
+%#									if ( this.id == "sel-" + data.attachments[i].attid ) {
+%#										$("#"+this.id).val(data.attachments[i].node);
+%#										$("#"+this.id).trigger('change');
+%#									}
+%#								}
+%#							});
+						},
+						buttons: [
+% if (RT->Config->Get('AttacmentsTicketFeature', $session{'CurrentUser'})) {
+							{
+							    text: "<&|/l&>Add to new ticket</&>",
+								action: function( e, t, node, config ) {
+									var links = [];
+									// Use selectedRows instead of visible checkboxes
+									for (var attid in selectedRows) {
+										links.push(attid);
+									}
+									if (links.length === 0) return;
+									console.log(links.join(','));
+									$('#myModal').appendTo("body").modal('show');
+									$('#send1').off('click.attachmentsColored').on('click.attachmentsColored', function() {
+										$.ajax({
+										url: '<%$RT::WebPath%>/Ajax/AddNewTicket',
+										type: 'POST',
+										data: { 'submit':true, 'subj': $('#subj').val(), 'queue': $('select[name="Queue1"]').val(), 'owner': jQuery('#Owner1').val(), 'message': $('#message-text1').val(), 'attlist': links.join(',')  }, // An object with the key 'submit'
+										success: function (result) {
+											$.jGrowl(result, { themeState: 'none' });
+											location.reload();
+										}
+										});
+										$('#myModal').appendTo("body").modal('hide');
+									});
+							    }
+							},
+							{
+							    text: "<&|/l&>Add to existing ticket</&>",
+								action: function( e, t, node, config ) {
+									var links = [];
+									// Use selectedRows instead of visible checkboxes
+									for (var attid in selectedRows) {
+										links.push(attid);
+									}
+									if (links.length === 0) return;
+									console.log(links.join(','));
+									$('#myModal2').appendTo("body").modal('show');
+									$('#send2').off('click.attachmentsColored').on('click.attachmentsColored', function() {
+										$.ajax({
+										url: '<%$RT::WebPath%>/Ajax/AddToTicket',
+										type: 'POST',
+										data: { 'submit':true, 'id': $('#ticketid').val(), 'message': $('#message-text2').val() + '<br />', 'attlist': links.join(',') }, // An object with th
+										success: function (result) {
+											$.jGrowl(result, { themeState: 'none' });
+											location.reload();
+										}
+										});
+										$('#myModal2').appendTo("body").modal('hide');
+									});
+							    }
+							},
 % }
+							{
+							    text: "<&|/l&>Download</&>",
+									action: function( e, t, node, config ) {
+
+										var links = [];
+										var data = dt.rows({selected:  true}).data();
+										var zip = new JSZip();
+										var count = 0;
+										var zipFilename = "doxAttachments.zip";
+										// Use selectedRows instead of visible checkboxes
+										for (var attid in selectedRows) {
+											links.push(selectedRows[attid].link);
+										}
+%#									for (i=0; i<data.length; i++) {
+%#									    links.push(data[i].checkbox);
+%#									}
+									console.log(links);
+									switch (true) {
+										case (2 <= links.length):
+											links.forEach(function(url, i) {
+												var filename = url.split('/').pop();
+												JSZipUtils.getBinaryContent(url, function (err, dat) {
+												if (err) {
+													throw err; // or handle the error
+												}
+												zip.file(filename, dat, { binary: true });
+												count++;
+												if (count == links.length) {
+													zip.generateAsync({ type: 'blob' }).then(function (content) {
+													saveAs(content, zipFilename);
+													// Clear selections after successful download
+													selectedRows = {};
+													$('input[name="att"]:checkbox').prop('checked', false);
+													$('#checkAll').prop('checked', false).prop('indeterminate', false);
+													});
+												}
+												});
+											});
+											break;
+										case (1 == links.length):
+											const filename = links[0].split('/').pop();
+											fetch(links[0])
+												.then(response => response.blob())
+													.then(blob => {
+														const url = window.URL.createObjectURL(blob);
+														const a = document.createElement('a');
+														a.href = url;
+														a.download = filename;
+														a.click();
+														window.URL.revokeObjectURL(url);
+														// Clear selections after successful download
+														selectedRows = {};
+														$('input[name="att"]:checkbox').prop('checked', false);
+														$('#checkAll').prop('checked', false).prop('indeterminate', false);
+													});
+											break;
+										default: 
+											console.log('No links selected');
+									}
+								}
+							},
+% my $showUploadBtn = RT->Config->Get('ShowUploadButton');
+% my $anyUploadEnabled = ($SimpleUpload && $SimpleUpload->{'enabled'}) || ($GoogleDrive && $GoogleDrive->{'enabled'}) || ($OneDrive && $OneDrive->{'enabled'}) || ($NextCloud && $NextCloud->{'enabled'});
+% if ($showUploadBtn && $anyUploadEnabled) {
+							{
+%if ($SimpleUpload && $SimpleUpload->{'enabled'}) {
+							    text: "<&|/l&>Upload</&>",
+								action: function( e, t, node, config ) {
+									var uploadUrl = '<%RT->Config->Get('WebPath')%>/Ajax/AjaxUploadAttachment?TicketId=<%$Ticket->Id%>';
+									var uploadInput = $('<input type="file" multiple />');
+									uploadInput.on('change', function() {
+										var files = this.files;
+										if (files.length > 0) {
+											var formData = new FormData();
+											for (var i = 0; i < files.length; i++) {
+												formData.append('attachments[]', files[i]);
+											}
+											$.ajax({
+												url: uploadUrl,
+												type: 'POST',
+												data: formData,
+												contentType: false,
+												processData: false,
+												success: function(response) {
+													$.jGrowl(response.message, { themeState: 'success' });
+													location.reload();
+												},
+												error: function(xhr) {
+													$.jGrowl(xhr.responseText, { themeState: 'error' });
+												}
+											});
+										}
+									});
+									uploadInput.click();
+								}
+% } elsif ($GoogleDrive && $GoogleDrive->{'enabled'}) {
+	// Google Drive upload logic here
+								text: "<&|/l&>Upload</&> GDrive",
+								action: function( e, t, node, config ) {
+									var uploadUrl = '<%RT->Config->Get('WebPath')%>/Ajax/AjaxUploadGDrive?TicketId=<%$Ticket->Id%>';
+									var atts = [];
+									// Use selectedRows instead of visible checkboxes
+									for (var attid in selectedRows) {
+										atts.push(attid);
+									}
+									console.log(JSON.stringify(atts));
+									$.ajax({
+										url: uploadUrl,
+										type: 'POST',
+										data: {'attachments': JSON.stringify(atts)},
+										success: function(response) {
+											$.jGrowl(response.message, { themeState: 'success' });
+											location.reload();
+										},
+										error: function(xhr) {
+											$.jGrowl(xhr.responseText, { themeState: 'error' });
+										}
+									});
+								}
+
+% } elsif ($OneDrive && $OneDrive->{'enabled'}) {
+	// OneDrive upload logic here
+								text: "<&|/l&>Upload</&> OneDrive",
+								action: function( e, t, node, config ) {
+									var uploadUrl = '<%RT->Config->Get('WebPath')%>/Ajax/AjaxUploadOneDrive?TicketId=<%$Ticket->Id%>';
+									var atts = []; 
+									// Use selectedRows instead of visible checkboxes
+									for (var attid in selectedRows) {
+										atts.push(attid);
+									}
+									console.log(JSON.stringify(atts));
+									console.log(uploadUrl);
+
+ 									$.ajax({
+										url: uploadUrl,
+										type: 'POST',
+										data: {'attachments': JSON.stringify(atts)},
+										success: function(response) {
+											$.jGrowl(response.message, { themeState: 'success' });
+											location.reload();
+										},
+										error: function(xhr) {
+											$.jGrowl(xhr.responseText, { themeState: 'error' });
+										}
+									});
+
+								}
+%} elsif ($NextCloud && $NextCloud->{'enabled'}) {
+	// NextCloud upload logic here
+								text: "<&|/l&>Upload</&> NextCloud",
+								action: function( e, t, node, config ) {
+									var uploadUrl = '<%RT->Config->Get('WebPath')%>/Ajax/AjaxUploadNextCloud?TicketId=<%$Ticket->Id%>';
+									var atts = []; 
+									// Use selectedRows instead of visible checkboxes
+									for (var attid in selectedRows) {
+										atts.push(attid);
+									}
+									console.log(JSON.stringify(atts));
+									console.log(uploadUrl);
+
+ 									$.ajax({
+										url: uploadUrl,
+										type: 'POST',
+										data: {'attachments': JSON.stringify(atts)},
+										success: function(response) {
+											$.jGrowl(response.message, { themeState: 'success' });
+											location.reload();
+										},
+										error: function(xhr) {
+											$.jGrowl(xhr.responseText, { themeState: 'error' });
+										}
+									});
+
+								}
+%}
+							}
+% }
+						],
 
-% $m->callback( %ARGS, CallbackName => 'AfterList', TicketObj => $Ticket, Attachments => $Attachments, Documents => \%documents, IsChecked => \%is_checked, ShowMore => \$show_more );
+						data: displayedAttachments,
+						drawCallback: function( row, dataIndex ) {
+							$('#attachments input[name="att"]').each(function() {
+								$(this).prop('checked', !!selectedRows[this.value]);
+							});
+							if (dt) syncAttachmentSelection();
+							$('.selectnode').select2({
+								placeholder: '<&|/l&>Select an option</&>',
+								width: "230px",
+								data: data.folders,
+								formatSelection: function(item) {
+									return item.text
+								},
+								formatResult: function(item) {
+									return item.text
+								},
+								templateResult: formatResult,
+								dropdownCssClass : 'bigdrop',
+								dropdownAutoWidth : true
+							});
+						},
+						createdRow: function( row, data, dataIndex ) {
+							if (data['category'] != 'X') {
+								if (!isClassDefined('darkmode')) {
+									$(row).css( 'background-color', colors[data['category']] );
+								} else {
+									$(row).find('td').attr('style', 'background:' + colors[data['category']] + ' !important');
+								}
+								$(row).css( 'background-color', colors[data['category']] );
+							};
+							if ( (data['category'] == "hidden" ) && ( $('#showhide').val() != '1') ) {
+								if (!isClassDefined('darkmode')) {
+									$(row).css( 'background-color', '#DCDCDC' );
+								} else {
+									$(row).find('td').attr('style', 'background:#DCDCDC !important');
+								}
+								$(row).hide();
+							};
+						},
+					columns: [
+						{ "data": null, "defaultContent": "","orderable": false,
+						"fnCreatedCell": function (nTd, sData, oData, iRow, iCol) {
+							var checkbox = $('<input>').attr({ type: 'checkbox', name: 'att', 'data-link': oData.checkbox }).val(oData.attid);
+							checkbox.prop('checked', !!selectedRows[oData.attid]);
+							checkbox.on('change', function() { handleCheckboxChange(this); });
+							$(nTd).empty().append(checkbox);
+						}},
+							{ "data":"category",
+		                    	//"orderable": false,
+								"fnCreatedCell": function (nTd, sData, oData, iRow, iCol) {
+									var rgroup = "";
+									for (var key in colors) {
+										rgroup += "<label class='__container'>";
+										rgroup += "<input type='radio' onchange='setCategory(" + iRow + ","  + iCol + ", this.name, this.value);' name='rad" + oData.attid + "' id='rad-" + oData.attid + "-" + key + "' value='" + key +"'";
+										rgroup += (key == sData)?" checked":"";
+										rgroup += " />";
+										rgroup += "<span class='checkmark' style='background-color: " + colors[key] + " !important' title='" + key + "'></span>";
+										rgroup += "</label>";
+									}
+
+									rgroup += "</label>";
+									rgroup += "<label class='__container'>";
+//									rgroup += "id='del-" + oData.attid + "-label'";
+//									rgroup += (sData == 'X')?" style='display:none;'>":">";
+									// &#x21bb;
+									// &#x238C;
+									rgroup += "<span title='<%loc('Unhighlight')%>' id='del" + oData.attid + "' class='checkmark' style='background:#FFF !important;text-align:center;color:#000 !important;' onclick='delCategory("+iRow+", this.id); return true;'>&nbsp;</span>";
+									rgroup += "</label>";
+
+									//&#x2612;
+									rgroup += "<label class='__container'>";
+									rgroup += "<span title='<%loc('Hide')%>' id='hidden-" + oData.attid + "' onclick='setCategory(" + iRow + ","  + iCol + ", \"rad" + oData.attid + "\", \"hidden\");' class='checkmark' style='background:#DCDCDC !important;text-align:center;";
+									rgroup += (sData == "hidden")?"color:#000 !important;":"color:#000 !important;";
+									rgroup += "'>&#x2715;</span>";
+									rgroup += "<input type='hidden' name='rad" + oData.attid + "' id='rad-" + oData.attid + "-hidden' value='";
+									rgroup += (sData == "hidden")?"hidden":"";
+									rgroup += "'>";
+									$(nTd).css({ 'min-width': '80px', 'max-width': '120px' });
+									$(nTd).html( rgroup );
+									return oData.attid;
+								}
+							},
+							{ "data":"filename",
+							"fnCreatedCell": function (nTd, sData, oData, iRow, iCol) {
+								var showFull = <% RT->Config->Get('ShowFullFilename', $session{'CurrentUser'}) ? 'true' : 'false' %>;
+								if (showFull) {
+									$(nTd).addClass('__fullFilename');
+									$(nTd).html(sData);
+								} else {
+									$(nTd).css({'white-space': 'nowrap', 'overflow': 'hidden', 'text-overflow': 'ellipsis', 'max-width': '200px'});
+									$(nTd).attr('title', sData);
+								}
+								if (!oData.is_uri_list) {
+									var attachmentUrl = <% RT->Config->Get('WebPath') . '/' . $DisplayPath . '/Attachment/' |n,j %> + oData.trid + '/' + oData.attid + '/' + encodeURIComponent(oData.raw_filename);
+									$(nTd).find('a').attr('href', attachmentUrl);
+									$(nTd).find('img').attr('src', attachmentUrl);
+								}
+							}
+						},
+							{ "data":"date_unix",
+								"fnCreatedCell": function (nTd, sData, oData, iRow, iCol) {
+									$(nTd).html( oData.date );
+								}
+							},
+							{ "data":"creator" },
+							{ "data":"size" },
+							{ "data":"node" ,
+								"fnCreatedCell": function (nTd, sData, oData, iRow, iCol) {
+									var folderSelect = $('<select>').addClass('selectnode').attr('id', 'sel-' + oData.attid);
+									folderSelect.append($('<option>'));
+									folderSelect.on('change', function() { setFolder(this.id, this.value); });
+									data.folders.forEach((element) => {
+										folderSelect.append($('<option>').val(element.id).text(element.text).prop('selected', element.id == oData.node));
+									});
+									$(nTd).empty().append(folderSelect);
+
+								}
+							},
+							{ "data":"comment", "render": $.fn.dataTable.render.text(),
+								"fnCreatedCell": function (nTd, sData, oData, iRow, iCol) {
+									renderAttachmentComment(nTd, sData, oData.attid);
+								}
+							}],
+						autoWidth: false,
+						scrollX: false,
+						columnDefs: [
+								{ type: 'file-size', targets: 5 },
+								{ width: '30px', targets: 0, className: 'dt-nowrap' },
+								{ width: '120px', targets: 1, className: 'dt-nowrap' },
+								{ width: '80px', targets: 3, className: 'dt-nowrap' },
+								{ width: '90px', targets: 4, className: 'dt-nowrap' },
+								{ width: '60px', targets: 5, className: 'dt-nowrap' },
+								{ width: '150px', targets: 6, className: 'dt-nowrap' },
+								{ width: '120px', targets: 7, className: 'dt-nowrap' }
+							],
+		                order: [ 3, 'asc' ],
+					});
+					syncAttachmentSelection();
+					if (displayedAttachments.length < allAttachments.length) {
+						var showAllButton = $('<button>').attr('type', 'button').addClass('button btn btn-primary').text(<% loc('Show all') |n,j %>);
+						showAllButton.on('click', function() {
+							dt.clear().rows.add(allAttachments).draw(false);
+							showAllButton.remove();
+						});
+						$('.attachments-table-wrapper').append(showAllButton);
+					}
+				}
+			}).then(function (data) {
+			    // Event delegation for dynamically created elements
+				$(document).on('mouseenter', '.__dropdown', function() {
+					const dropdownContent = $(this).find('.__dropdown-content');
+					const rect = $(this)[0].getBoundingClientRect();
+					const viewportHeight = $(window).height();
+					const viewportWidth = $(window).width();
+
+					// Remove any existing positioning classes
+					dropdownContent.removeClass('above below');
+
+					// Calculate horizontal position - align left edge with element, but don't overflow right side
+					let leftPos = rect.left;
+					const dropdownWidth = 480; // max-width from CSS
+					if (leftPos + dropdownWidth > viewportWidth) {
+						leftPos = Math.max(10, viewportWidth - dropdownWidth - 10);
+					}
+
+					// Check if the element is in the upper half of the screen
+					if (rect.top < viewportHeight / 2) {
+						// Position below the element
+						dropdownContent.addClass('below');
+						dropdownContent.css({
+							'top': rect.bottom + 'px',
+							'bottom': 'auto',
+							'left': leftPos + 'px'
+						});
+					} else {
+						// Position above the element
+						dropdownContent.addClass('above');
+						dropdownContent.css({
+							'top': 'auto',
+							'bottom': (viewportHeight - rect.top) + 'px',
+							'left': leftPos + 'px'
+						});
+					}
+
+					// Ensure the dropdown is visible
+					dropdownContent.css('display', 'block');
+				});
+
+			    // Hide dropdown when mouse leaves the dropdown area
+				$(document).on('mouseleave', '.__dropdown', function() {
+					$(this).find('.__dropdown-content').css('display', 'none');
+				});
+
+				// Hide dropdown when mouse leaves the dropdown area
+				$('.__dropdown').on('mouseleave', function() {
+					$(this).find('.__dropdown-content').css('display', 'none');
+				});
+			});
+		});
+    });
+
+    function setCategory( row, cell, radio, value ) {
+		console.log(row, cell, radio, value);
+		if (!isClassDefined('darkmode')) {
+			jQuery(dt.row(row).node()).css( 'background', colors[value] );
+		} else {
+			jQuery(dt.row(row).node()).find('td').attr('style', 'background:' + colors[value] + ' !important');
+		}
+		if (value == 'hidden') {
+			jQuery( dt.row(row).node() ).find("input[type='hidden'][name^='rad']").val('hidden');
+			if (!isClassDefined('darkmode')) {
+				jQuery(dt.row(row).node()).css( 'background', '#DCDCDC' );
+			} else {
+				jQuery(dt.row(row).node()).find('td').attr('style', 'background:#DCDCDC !important');
+			}
+			if (!jQuery("#showhide").is(":checked")) {
+				jQuery(dt.row(row).node()).hide();	
+			}
+		};
+		jQuery.ajax({
+			type: 'POST',
+			url: '<%RT->Config->Get('WebPath')%>/Ajax/AjaxModifyCategory',
+			data: {
+				'Value' : value,
+				'attachId' : radio.replace("rad", ""),
+			},
+			success:function(){
+				jQuery("#del-"+radio.replace("rad", "")+"-label").show();
+				console.log("Category set to: " + value);
+			}
+		});
+    };
+
+    function delCategory( row, radio ) {
+		jQuery.ajax({
+			type: 'POST',
+			url: '<%RT->Config->Get('WebPath')%>/Ajax/AjaxDeleteCategory',
+			data: {
+				'attachId'	: radio.replace("del", ""),
+			},
+			success:function(){
+				var r = radio.replace("del", "rad")
+				jQuery("[name="+r+"]").prop('checked', false);
+				jQuery("#del-"+radio.replace("del", "")+"-label").hide();
+				jQuery( dt.row(row).node() ).find("input[type='hidden'][name="+r+"]").removeAttr('value');
+				if (!isClassDefined('darkmode')) {
+					jQuery(dt.row(row).node()).removeAttr('style');
+				} else {
+					jQuery(dt.row(row).node()).find('td').removeAttr('style');
+					jQuery(dt.row(row).node()).css( 'background', '' );
+				}
+				console.log(row, r, radio);
+			}
+		});
+    };
+
+	function setFolder( attID, node ) {
+		jQuery.ajax({
+			type: 'POST',
+			url: '<%RT->Config->Get('WebPath')%>/Ajax/AjaxModifyFolder',
+			data: {
+				'Value' : node,
+				'attachId' : attID.replace("sel-", "")
+			}
+		});
+	}
+
+	function hiddenShowHide( checked ) {
+		dt.rows().every( function ( index ) {
+			if (checked) {
+				if (!jQuery(dt.row(index).node()).is(':visible')){
+					jQuery(dt.row(index).node()).show();
+				}
+			} else {
+				if (jQuery( dt.row(index).node() ).find("input[type='hidden'][name^='rad']").val() == 'hidden') {
+					jQuery(dt.row(index).node()).hide();
+								}
+								}
+		});
+	}
+
+	function renderAttachmentComment(cell, value, attachmentId) {
+		var commentText = value == null ? '' : String(value);
+		var label = document.createElement('span');
+		label.className = '__attachmentName inverted-text';
+		label.style.display = 'none';
+		label.textContent = commentText;
+
+		var input = document.createElement('input');
+		input.className = 'comment-input';
+		input.style.borderStyle = 'none';
+		input.style.padding = '5px';
+		input.id = 'comm-' + attachmentId;
+		input.type = 'text';
+		input.value = commentText;
+		input.disabled = true;
+		input.addEventListener('focusout', function() { saveComment(this.id, this.value); });
+
+		var editIcon = document.createElement('span');
+		editIcon.className = 'fas fa-pencil-alt icon-bordered fa-2x inverted-text';
+		editIcon.style.verticalAlign = 'middle';
+		editIcon.addEventListener('click', function() { editComment(input.id); });
+
+		cell.textContent = '';
+		cell.appendChild(label);
+		cell.appendChild(input);
+		cell.appendChild(document.createTextNode('\u00a0 '));
+		cell.appendChild(editIcon);
+	}
+
+	function editComment( editID ) {
+		jQuery( "#"+editID ).prop( "disabled", false ).focus();
+	}
+
+	function saveComment ( editID, editVal ) {
+		jQuery( "#"+editID ).prop( "disabled", true );
+		jQuery.ajax({
+			type: 'POST',
+			url: '<%RT->Config->Get('WebPath')%>/Ajax/AjaxModifyComment',
+			data: {
+				'Value' : editVal,
+				'attachId' : editID.replace("comm-", "")
+			}
+		});
+	}
+
+	function copyClipboard( aID ) {
+	/* Get the text field */
+		var copyText = document.getElementById(aID).getAttribute("href").replace('file:///','');
+		/* Select the text field */
+		document.addEventListener('copy', function(e) {
+			e.clipboardData.setData('text/plain', copyText);
+			e.preventDefault();
+		}, true);
+
+		/* Copy the text inside the text field */
+		document.execCommand("copy");
+
+		/* Alert the copied text */
+		alert("Copied the text: " + copyText);
+	}
+
+	function sendOpenReqMQ( aID ) {
+	/* Get the text field */
+		var copyText = document.getElementById(aID).getAttribute("href").replace('file:///','');
+		jQuery.ajax({
+			type: 'POST',
+			url: '<%RT->Config->Get('WebPath')%>/Ajax/AjaxSendMQOpen',
+			data: {
+				'queue' : '<% $session{'CurrentUser'}->Name %>',
+				'attachment' : copyText
+			}
+		}).done(function( data ) {
+			console.log( "Sent2MQ: " + data );
+		});
+		
+	}
+
+    function handleCheckboxChange(checkbox) {
+        var attid = checkbox.value;
+        var link = checkbox.getAttribute('data-link');
+        
+        if (checkbox.checked) {
+            selectedRows[attid] = {
+                attid: attid,
+                link: link
+            };
+        } else {
+            delete selectedRows[attid];
+        }
+        
+		syncAttachmentSelection();
+	}
+
+	function syncAttachmentSelection() {
+		if (selectableAttachments) {
+			var container = document.getElementById('attachment-existing-selected');
+			container.textContent = '';
+			Object.keys(selectedRows).forEach(function(attachmentId) {
+				var input = document.createElement('input');
+				input.type = 'hidden';
+				input.name = 'AttachExisting';
+				input.value = attachmentId;
+				container.appendChild(input);
+			});
+		}
+		var totalRows = dt.rows().count();
+		var selectedCount = 0;
+		dt.rows().every(function() {
+			if (selectedRows[this.data().attid]) selectedCount++;
+		});
+        var selectAllCheckbox = document.getElementById('checkAll');
+        
+        if (selectedCount === 0) {
+            selectAllCheckbox.checked = false;
+            selectAllCheckbox.indeterminate = false;
+        } else if (selectedCount === totalRows) {
+            selectAllCheckbox.checked = true;
+            selectAllCheckbox.indeterminate = false;
+        } else {
+            selectAllCheckbox.checked = false;
+            selectAllCheckbox.indeterminate = true;
+        }
+    }
 
-% if ($show_more) {
-<span class="show-more-link">
-% my %params = %ARGS;
-% delete $params{Ticket};
-% delete $params{Attachments};
-% delete $params{Count};
-% my $query = $m->comp('/Elements/QueryString', %params, id => $Ticket->id );
-% my $url   = RT->Config->Get('WebPath')."/Helpers/TicketAttachments?$query";
+    function checkAllClick(cb) {
+        // Clear or populate selectedRows based on check all state
+        if (cb.checked) {
+            // Select all rows
+            dt.rows().every(function() {
+                var data = this.data();
+                selectedRows[data.attid] = {
+                    attid: data.attid,
+                    link: data.checkbox
+                };
+            });
+        } else {
+            // Deselect all rows
+            selectedRows = {};
+        }
+        
+        // Update visible checkboxes
+		jQuery('#attachments input[name="att"]:checkbox').not(cb).prop('checked', cb.checked);
+		syncAttachmentSelection();
+    }
 
-<script type="text/javascript">
-    function showAllAttachments(node) {
-        var container = node.closest('.attachment-list');
-        var params = node.closest('form').find('input[name=AttachExisting]').serialize();
-
-        node.parent().text(<% loc('Loading...') | n,j%>);
-
-        var url = <% $url |n,j %>;
-        if (params) url += '&' + params;
-        container.load(url);
+    function newTicketClick() {
+        var links = [];
+        // Use selectedRows instead of visible checkboxes
+        for (var attid in selectedRows) {
+            links.push(attid);
+        }
+        if (links.length === 0) return;
+        console.log(links.join(','));
+        jQuery('#myModal').appendTo("body").modal('show');
+		jQuery('#send1').off('click.attachmentsColored').on('click.attachmentsColored', function() {
+            jQuery.ajax({
+                url: '<%$RT::WebPath%>/Ajax/AddNewTicket',
+                type: 'POST',
+                data: { 'submit':true, 'subj': jQuery('#subj').val(), 'queue': jQuery('select[name="Queue1"]').val(), 'owner': jQuery('#Owner1').val(), 'message': jQuery('#message-text1').val(), 'attlist': links.join(',')  }, // An object with the key 'submit'
+                success: function (result) {
+                    jQuery.jGrowl(result, { themeState: 'none' });
+                    location.reload();
+                }
+            });
+            jQuery('#myModal').appendTo("body").modal('hide');
+        });
     }
-</script>
 
-<a href="#" class="button btn btn-primary" onclick="showAllAttachments(jQuery(this)); return false;" ><% loc('Show all') %></a>
+    function okTicketClick() {
+        var links = [];
+        // Use selectedRows instead of visible checkboxes
+        for (var attid in selectedRows) {
+            links.push(attid);
+        }
+        if (links.length === 0) return;
+        console.log(links.join(','));
+        jQuery('#myModal2').appendTo("body").modal('show');
+		jQuery('#send2').off('click.attachmentsColored').on('click.attachmentsColored', function() {
+            jQuery.ajax({
+                url: '<%$RT::WebPath%>/Ajax/AddToTicket',
+                type: 'POST',
+                data: { 'submit':true, 'id': jQuery('#ticketid').val(), 'message': jQuery('#message-text2').val() + '<br />', 'attlist': links.join(',') }, // An object with th
+                success: function (result) {
+                    jQuery.jGrowl(result, { themeState: 'none' });
+                    location.reload();
+                }
+            });
+            jQuery('#myModal2').appendTo("body").modal('hide');
+        });
+    }
 
-</span>
-% }
+    function isClassDefined(className) {
+	for (let sheet of document.styleSheets) {
+	    try {
+		for (let rule of sheet.cssRules || []) {
+		    if (rule.selectorText && rule.selectorText.includes('.' + className)) {
+			return true;
+		    }
+		}
+	    } catch (e) {
+	// Ignore cross-origin stylesheets
+	    }
+	}
+	return false;
+    }
+</script>
 
-</div>
 </&>
 
 <%INIT>
+use utf8;
+use JSON;
+
+my $SimpleUpload = RT->Config->Get('SimpleUpload');
+my $GoogleDrive = RT->Config->Get('GoogleDrive');
+my $OneDrive = RT->Config->Get('OneDrive');
+my $NextCloud = RT->Config->Get('NextCloud');
 
 # If we haven't been passed in an Attachments object (through the precaching mechanism)
 # then we need to find one
@@ -138,18 +978,19 @@
 
 my $show_more = 0;
 my %documents;
+my %colormap = RT->Config->Get('AttachmentCategories');
 
 # show newest first
 $Attachments->OrderByCols(
-    { FIELD => 'Created', ORDER => 'DESC' },
-    { FIELD => 'id',      ORDER => 'DESC' },
+	{ FIELD => 'Created', ORDER => 'DESC' },
+	{ FIELD => 'id',      ORDER => 'DESC' },
 );
 
 while ( my $attach = $Attachments->Next() ) {
    # display "show more" only when there will be more attachments
    if (defined($Count) && --$Count < 0) {
-       $show_more = 1;
-       last;
+	   $show_more = 1;
+	   last;
    }
    push @{ $documents{ $attach->Filename } }, $attach;
 }
--- share/html/Ticket/Elements/ShowSummary
+++ local/html/Ticket/Elements/ShowSummary
@@ -2,7 +2,7 @@
 %#
 %# COPYRIGHT:
 %#
-%# This software is Copyright (c) 1996-2026 Best Practical Solutions, LLC
+%# This software is Copyright (c) 1996-2021 Best Practical Solutions, LLC
 %#                                          <sales@bestpractical.com>
 %#
 %# (Except where explicitly superseded by other copyright notices)
@@ -92,7 +92,8 @@
         title_href   => ($can_modify || $can_modify_cf) ? RT->Config->Get('WebPath')."/Ticket/Modify.html" : "",
         InlineEdit   => ($can_modify || $can_modify_cf) ? $InlineEdit : 0,
         &>
-% $m->callback( %ARGS, CallbackName => 'AfterCustomFields' );
+    % $m->callback( %ARGS, CallbackName => 'AfterCustomFields' );
+
 <%PERL>
 my $people_url = RT->Config->Get('WebPath')."/Ticket/ModifyPeople.html?id=".$Ticket->Id;
 my $people_inline = sprintf( $modify_inline, $m->interp->apply_escapes( $people_url, 'h' ) );
@@ -123,8 +124,7 @@
 </&>
 
 % $m->callback( %ARGS, CallbackName => 'AfterPeople' );
-    <& /Ticket/Elements/ShowAttachments, Ticket => $Ticket, Attachments => $Attachments, Count => RT->Config->Get('AttachmentListCount') &>
-% $m->callback( %ARGS, CallbackName => 'AfterAttachments' );
+
     <& /Ticket/Elements/ShowRequestor, Ticket => $Ticket &>
 % $m->callback( %ARGS, CallbackName => 'LeftColumn' );
   </div>
@@ -142,7 +142,10 @@
 % }
 % $m->callback( %ARGS, CallbackName => 'AfterReminders' );
 
+% if ( RT->Config->Get('AttachmentsColoredShowArticle') ) {
 <& /Elements/ShowArticle, Ticket => $Ticket &>
+% }
+
 <%PERL>
 my $dates_url = RT->Config->Get('WebPath')."/Ticket/ModifyDates.html?id=".$Ticket->Id;
 my $dates_inline = sprintf( $modify_inline, $m->interp->apply_escapes( $dates_url, 'h' ) );
@@ -193,6 +196,8 @@
 push @extra, (titleright_raw => $links_titleright) if $links_titleright;
 </%PERL>
 % $m->callback( %ARGS, CallbackName => 'LinksExtra', extra => \@extra );
+  </div>
+<div class="boxcontainer col-md-12" style="padding-left:15px;margin-top:7px;">
     <&| /Widgets/TitleBox, title => loc('Links'),
         ($can_modify ? (title_href => $links_url) : ()),
         class => (join " ", 'ticket-info-links', ($links_behavior eq 'always' ? 'editing' : ())),
@@ -211,7 +216,8 @@
                 <& /Elements/EditLinks, Object => $Ticket, TwoColumn => 0 &>
 
                 <h3><&|/l&>Merge</&></h3>
-                <& /Ticket/Elements/EditMerge, Ticket => $Ticket, MergeTextClass => '', %ARGS &>
+                <& /Ticket/Elements/EditMerge, Ticket => $Ticket, LabelStyle => 'col-3',
+                   ValueStyle => 'col-9', MergeTextClass => '', %ARGS &>
 
                 <div class="form-row">
                   <div class="col-12 text-right">
@@ -224,6 +230,14 @@
 % $m->callback( %ARGS, CallbackName => 'RightColumn' );
   </div>
 </div>
+
+<div class="row">
+    <div class="boxcontainer col-md-12">
+        <& /Ticket/Elements/ShowAttachments, Ticket => $Ticket, Attachments => $Attachments, Count => RT->Config->Get('AttachmentListCount') &>
+% $m->callback( %ARGS, CallbackName => 'AfterAttachments' );
+    </div>
+</div>
+
 <%ARGS>
 $Ticket => undef
 $Attachments => undef
